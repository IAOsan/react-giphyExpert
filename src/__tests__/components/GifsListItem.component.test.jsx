import 'vitest';
import '@testing-library/jest-dom';
import { render } from '../test-utils';
import GifsListItem from '../../components/GifsListItem.component';

function renderItem(props = {}) {
	return render(<GifsListItem {...props} />);
}

describe('<GifsListItem />', () => {
	const title = 'testing image';
	const images = {
		downsized: {
			url: 'https://dummyimage.com/360x360/fff/aaa',
		},
	};
	let wrapper = renderItem({
		title,
		images,
	});

	beforeEach(() => {
		wrapper = renderItem({
			title,
			images,
		});
	});

	it('should render element properly', () => {
		expect(wrapper.container.firstElementChild).toMatchSnapshot();
	});
	it('should image contains an src with the image url and alt with title provided', () => {
		expect(wrapper.getByRole('img')).toHaveAttribute('alt', title);
		expect(wrapper.getByRole('img')).toHaveAttribute(
			'src',
			images.downsized.url
		);
	});
	it('should image contains fluid classname', () => {
		expect(wrapper.getByRole('img')).toHaveClass('img-fluid');
	});
});
