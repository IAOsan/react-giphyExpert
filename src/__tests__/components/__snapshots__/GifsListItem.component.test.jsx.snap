// Vitest Snapshot v1

exports[`<GifsListItem /> > should render element properly 1`] = `
<div
  class="col-lg-4 col-md-2"
>
  <img
    alt="testing image"
    class="img-fluid my-2"
    src="https://dummyimage.com/360x360/fff/aaa"
  />
</div>
`;
